### ISSUE TITLE
Class-wise mAP only showing results for a few classes during evaluation

### ISSUE THREAD

**Issue by gauricollab09** [PaddlePaddle/PaddleDetection#9182]
### 问题确认 Search before asking

- [X] 我已经搜索过问题，但是没有找到解答。I have searched the question and found no related answer.

### 请提出你的问题 Please ask your question

Hi all,

I'm encountering an issue while evaluating my object detection model that I trained using the `--classwise` argument in the command as follows: 
`python3 tools/eval.py -c configs/ppyoloe/voc/ppyoloe_plus_crn_s_30e_voc_NC.yml --classwise --amp`
![image](https://github.com/user-attachments/assets/2b30fac4-4c36-4259-b672-5887c751fc5b)

The evaluation only displays the AP for the first two classes in the label_list.txt, but I have multiple classes in my dataset. Here are the details of my setup:

Environment:

PaddleDetection version: 3.0.0-betal
PaddlePaddle version: release/2.5
OS: Ubuntu 20.04

What I have tried so far:

I verified that my dataset contains annotations for all 19 classes.
I used a lower confidence threshold (--conf_thres 0.001) to include more detections, but the issue persists.
I checked the label_list.txt. 

Could someone else me with this? Thank you!

**Comment by liu-jiaxuan** [comment]
Hi, please check if other output results (e.g., PR curve) include all categories

**Comment by gauricollab09** [comment]
@liu-jiaxuan  It contains the curves for only the two categories - cup and person.

**Comment by liu-jiaxuan** [comment]
Hi, the PaddleDetection beta version may have potential problems. Please use PaddleX first. You can refer to the [tutorial](https://github.com/PaddlePaddle/PaddleX/blob/release/3.0-beta1/docs/module_usage/tutorials/cv_modules/object_detection_en.md#43-model-evaluation)

**Comment by TingquanGao** [comment]
The issue has no response for a long time and will be closed. You can reopen or new another issue if are still confused.

---
_From Bot_

**Comment by gauricollab09** [comment]
Hi, sorry for opening the same issue. 
I have been investigating this issue and one thing I found was that the names of my images had some emojis, spaces etc. Because of this only images of 3 classes were being given to the model. I cleaning all names, I made sure the classes given to my model had all the 19 classes that I want to train. However after this, I trained the model and tried the same script for evaluation. And I managed to get results for 7 classes out of 19. There are also some None classes. I am unable to understand why this is happening. Can someone give me some pointers on how I can look at this problem and think of what is wrong?
![image](https://github.com/user-attachments/assets/d4b7b1f7-a39e-4890-b139-e01b9ec5a168)

Also there is another observation I have:
   - When running inference script (`tools/infer.py`): Model detects objects across multiple classes (there are Phone instances also detected), but 
   - When running evaluation script (`tools/eval.py`): Only 7 classes show AP metrics, others are "None"

Additional information : 
- Model: PPYOLOE_plus_crn_l
- Number of classes: 19
